app/routes/__auth/user.server.ts

```diff
@@ -14,8 +15,18 @@ export async function getUser({ headers }: Request) {
   const email = session.get('email') as string
   const groups = session.get('groups') as string[]
   const idp = session.get('idp') as string | null
-
+  const refreshToken = session.get('refreshToken') as string
+  const cognitoUserName = session.get('cognitoUserName') as string
   if (!sub) return null
+  const user = { sub, email, groups, idp, refreshToken, cognitoUserName }
+  await getLatestUserGroups(user)
```

Comment:
Won't this _always_ submit the refresh token on each page load?

We should only submit the refresh token if (1) the access token is absent or (2) the access token has expired.

Reply:
Just to clarify on 1, if the access token is absent or if the permission is absent?

The access token should be there either way, right?

Reply:
Doesn't matter, it's a false alarm. We aren't using this as a URL. I had previously dismissed the alert.